src/config/path.zig

```diff
@@ -0,0 +1,470 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const assert = std.debug.assert;
+const Allocator = std.mem.Allocator;
+const ArenaAllocator = std.heap.ArenaAllocator;
+
+const cli = @import("../cli.zig");
+const internal_os = @import("../os/main.zig");
+const formatterpkg = @import("formatter.zig");
+
+const log = std.log.scoped(.config);
+
+pub const ParseError = error{ValueRequired} || Allocator.Error;
+
+/// Path is like a string that represents a path value. The difference is that
+/// when loading the configuration the value for this will be automatically
+/// expanded relative to the path of the config file (or the home directory).
+pub const Path = union(enum) {
+    /// Path is unset.
+    null,
+
```

Comment:
Why is `null` a variant? I'm pretty sure our config mechanism supports just setting a field to `?Path`

Reply:
Fixed.